File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        
        HashSet<Character> skipSet = new HashSet<>();
        for(int i=0; i<skip.length(); i++){
            skipSet.add(skip.charAt(i));
        }
        
        StringBuilder sb = new StringBuilder();
        
        for(int i=0; i<s.length(); i++){
            int charIndex = s.charAt(i) - 'a';
            int shiftCount = 0;
            
            while(shiftCount < index){
                charIndex = (charIndex+1)% 26;
                
                char nextChar = (char) (charIndex + 'a');
                
                if(skipSet.contains(nextChar)){
                    continue;
                }
                
                shiftCount++;
                
            }
            
            sb.append((char)(charIndex +'a'));
            
        }
        
        
        return sb.toString();
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    
    vector<char> alpabet;
    for(char i = 'a' ; i<='z'; i++){
        if(skip.find(i) == string::npos) {
            alpabet.push_back(i);
        }
    }
    
    for(char c: s){
        int currIndex = 0;
        for(int i=0; i<alpabet.size();i++){
            if(alpabet[i] == c){
                currIndex = i;
                break;
            }
        }
        
        int nextIndex = (currIndex + index) % alpabet.size();
        answer += alpabet[nextIndex];
    }
    
    return answer;
}